src/test/HostActivationTests/FrameworkResolution/FrameworkResolutionBase.cs

```diff
@@ -85,6 +85,34 @@ public TestSettings With(Func<TestSettings, TestSettings> modifier)
             }
         }
 
+        protected void RunTest(
+            DotNetCli dotnet,
+            TestApp app,
+            TestSettings settings,
+            string resolvedFrameworkName,
+            string resolvedFrameworkVersion,
```

Comment:
This seems to violate the intent of fluent assertions.  It seems like the result should be returned and tested locally. To make the test quickly readable.

If you are worried about verbosity, perhaps we could change 
`DidNotFindCompatibleFrameworkVersion()` to include .Should().Fail().  Effectively 
`Should().Fail().And.DidNotFindCompatibleFrameworkVersion()` becomes `DidNotFindCompatibleFrameworkVersion()`
And similarly 
`Should().Pass().And.HaveResolvedFramework(resolvedFrameworkName, resolvedFrameworkVersion);`
becomes
`HaveResolvedFramework(resolvedFrameworkName, resolvedFrameworkVersion);`

Reply:
Looking at the new tests, I at least see why this was done.  It make inline data easier...

Maybe something like

```C#
public static AndConstraint<CommandResultAssertions> ShouldHaveResolvedFramework(this CommandResultAssertions assertion, string resolvedFrameworkName, string resolvedFrameworkVersion)
{
                    if (resolvedFrameworkVersion != null)
                    {
                        result.Should().Pass()
                            .And.HaveResolvedFramework(resolvedFrameworkName, resolvedFrameworkVersion);
                    }
                    else
                    {
                        result.Should().Fail()
                            .And.DidNotFindCompatibleFrameworkVersion();
                    }
}
```

Reply:
Good point - I kind of forgot about fluent asserts... I'll see what can be done to make this:
* More in line with fluent asserts
* Still keep the tests short - as mostly they are just specifying the matrix with input and expected output.

Reply:
I went through all the tests, changed them to use fluent assertions instead (well at least it now reads like a test with a "Should" in there somewhere).
I also formatted the inline data into tables with headers.
I also added comments describing what the tests are trying to achieve.

Hopefully it makes them much more readable now.